src/internal_rules.jl

```diff
@@ -568,298 +573,258 @@ function EnzymeRules.reverse(
     return (nothing,)
 end
 
-function EnzymeRules.forward(
-    ::Const{typeof(cholesky)},
-    RT::Type{<:Union{Const, DuplicatedNoNeed, Duplicated}},
-    A::Union{Const, Duplicated};
-    kwargs...
-)
+function EnzymeRules.forward(::Const{typeof(cholesky)}, RT::Type, A; kwargs...)
     fact = cholesky(A.val; kwargs...)
     if RT <: Const
         return fact
-    end
-    # TODO: This will be a problem for sparse matrices as invL and dL are dense
-    invL = inv(fact.L)
-    # TODO: dL is dense even when L was sparse
-    dL = Matrix(fact.L * LowerTriangular(invL * A.dval * invL' * 0.5 * I))
-    # TODO: Stored as Cholesky, although it isn't a Cholesky factorization
-    dfact = Cholesky(dL, 'L', 0)
-    if RT <: DuplicatedNoNeed
-        return dfact
     else
-        return Duplicated(fact, dfact)
-    end
-end
+        N = width(RT)
 
-function EnzymeRules.forward(
-    ::Const{typeof(cholesky)},
-    RT::Type{<:Union{BatchDuplicatedNoNeed, BatchDuplicated}},
-    A::Union{BatchDuplicatedNoNeed{T,N}, BatchDuplicated{T,N}};
-    kwargs...
-) where {T,N}
-    fact = cholesky(A.val; kwargs...)
-    if RT <: Const
-        return fact
-    end
-    invL = inv(fact.L)
-    dfact = ntuple(
-        i-> Cholesky(
-            Matrix(fact.L * LowerTriangular(invL * A.dval[i] * invL' * 0.5 * I)), 'L', 0
-        ), Val(N)
-    )
-    if RT <: BatchDuplicatedNoNeed
-        return dfact
-    else
-        return BatchDuplicated(fact, dfact)
-    end
-end
+        invL = inv(fact.L)
 
-function EnzymeRules.forward(
-        func::Const{typeof(\)},
```

Comment:
It seems the PR removed rules for `\` - doesn't this break support for StaticArrays and other arrays that cannot be mutated in-place?